### PR TITLE
Add Maximum Subarray Sum Solution

### PR DESCRIPTION
**Pull request by jessechoe10**

### CODE REVIEW

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 1]
```diff
@@ -102,11 +102,98 @@ export const problems = {
...
+This problem has two types of solutions one of which is called the  [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). However, understanding Kadane's algorithm is **optional** for this problem, since it can also be solved using prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+<Spoiler title="Solution">
+
+### Solution
+
+Suppose that we store the prefix sums at each index in the array. Intuitively, the maximum subarray sum ending at each index is the difference between the maximum prefix sum and the minimum prefix sum. Thus, we can iterate each element in the array and store the minimum prefix sum up to the current index. Then, we can find the maximum subarray sum by subtracting the current prefix sum to the minimum prefix sum.
+
+C++ code:
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:113]: Remove or put this in the CPPSection

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 2]
```diff
@@ -102,11 +102,98 @@ export const problems = {
...
+
+<Spoiler title = "Alternative Solution">
+
+### Solution
+
+Suppose we fix the first element of a subarray. The Kadane's Algorithm solution involves starting a new subarray sum at an element if that element is greater than the current subarray sum with an addition to the current element. If the subarray sum ending at the current element is greater than the current element, then we continue the subarray sum. Otherwise, we set the new subarray sum equal to the current element and restart our subarray sum. To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$.
+
+Watch this [video](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s) for a more detailed explanation of the approach above.
+
+C++ code:
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:163]: Same here - remove or put this in the CPPSection

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 3]
```diff
@@ -102,11 +102,98 @@ export const problems = {
...
+int main(){
+	int n; cin >> n;
+	vl x(n);
+	for(int i=0; i<n; i++){
+		cin >> x[i];
+	}
+	ll current_sum = x[0], max_subarray_sum = x[0];
+	for(int i=1; i<n; i++){
+		current_sum = max(current_sum + x[i], x[i]); 
+	// Continue the subarray sum or start a new subarray sum beginning at the current element
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:185]: shorten or put this on two lines (also fix the indentation)

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply content/3_Silver/Prefix_Sums_2.mdx:185]: Yep

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 4]
```diff
@@ -102,11 +102,98 @@ export const problems = {
...
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+		// The max subarray sum is the maximum difference between two prefix sums
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:135]:
```suggestion
		// Max subarray sum is the maximum difference between two prefix sums
```
This is one character too long :(

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 5]
```diff
@@ -102,11 +102,98 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
-This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
+This problem has two types of solutions one of which is called the  [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). However, understanding Kadane's algorithm is **optional** for this problem, since it can also be solved using prefix sums.
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:105]: Original phrasing is better?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 6]
```diff
@@ -102,11 +102,98 @@ export const problems = {
...
+
+Note: It is recommended that you skip the following solution.
+
+</Warning>
+
+<Spoiler title = "Alternative Solution">
+
+### Solution
+
+Suppose we fix the first element of a subarray. The Kadane's Algorithm solution involves starting a new subarray sum at an element if that element is greater than the current subarray sum with an addition to the current element. If the subarray sum ending at the current element is greater than the current element, then we continue the subarray sum. Otherwise, we set the new subarray sum equal to the current element and restart our subarray sum. To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$.
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:159]: Not very fluid.

"if that element is greater than the current subarray sum with an addition to the current element" What does this mean?

"continue the subarray sum" what does this mean? Extend the current subarray to the right?

" To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$." is the easy part. You can just simplify it to "print the largest subarray sum obtained by the algorithm" or something

Comment (frodakcin) [reply content/3_Silver/Prefix_Sums_2.mdx:159]: Alternatively you can entirely waive the details by saying like "Kadane's Algorithm finds the largest subarray sum using a greedy method. Check the video below for details"

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 7]
```diff
@@ -102,11 +102,98 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
-This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
+This problem has two types of solutions one of which is called the  [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). However, understanding Kadane's algorithm is **optional** for this problem, since it can also be solved using prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+<Spoiler title="Solution">
+
+### Solution
```
[the remark this answers is not in the document]
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:107]: Ok.

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 3 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:185]: Should I remove the tab or should I add another tab??

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 8]
```diff
@@ -102,11 +102,96 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution
+
+<Spoiler title="Solution">
+
+Suppose that we store the prefix sums at each index in the array. Intuitively, the maximum possible $p[i]-p[j]$ where $j\leq i$. Thus, we can iterate each element in the array and store the minimum prefix sum up to the current index. Then, we can find the maximum subarray sum by taking the difference of the current prefix sum and the minimum prefix sum.
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:109]:
```suggestion
Consider the prefix sum array. The maximum subarray sum is the maximum possible $p[i]-p[j]$ where $j\leq i$. We can iterate over each index $i$ and keep a running minimum $p[j]$ value. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
```

I think this is more clear. Maybe play around w the wording to make it more understandable.

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 9]
```diff
@@ -102,11 +102,96 @@ export const problems = {
...
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+	
+	// Max subarray sum is the maximum difference between two prefix sums
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:134]: What's up with the spacing here?

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply content/3_Silver/Prefix_Sums_2.mdx:183]: Add another tab

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 9 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:134]: Should I delete it?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 10]
```diff
@@ -102,11 +102,94 @@ export const problems = {
...
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+			// Max subarray sum is the maximum difference between two prefix sums
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:133]: I thought the standard was to have comments be on the same indentation as the code?

Comment (frodakcin) [reply content/3_Silver/Prefix_Sums_2.mdx:133]: Looks fine tho

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 11]
```diff
@@ -102,11 +102,94 @@ export const problems = {
...
+int main(){
+	int n; cin >> n;
+	vl x(n);
+	for(int i=0; i<n; i++){
+		cin >> x[i];
+	}
+	ll current_sum = x[0], max_subarray_sum = x[0];
+	for(int i=1; i<n; i++){
+		current_sum = max(current_sum + x[i], x[i]); 
+	// Continue the subarray sum or start a new subarray sum beginning at the current element
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:181]: Yea fix this too

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 11 above)
Comment (frodakcin) [reply content/3_Silver/Prefix_Sums_2.mdx:181]: They  should be consistent at least

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 10 above)
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:133]:
```suggestion
		// Max subarray sum is the maximum difference between two prefix sums
```

**Comment by andrewwangva** [comment]
Wasn't there previously a solution to this focus problem? Anybody know what happened to it?

**Comment by frodakcin** [comment]
https://github.com/cpinitiative/usaco-guide/commit/832d4c6e2491e4040ba22323cda290719a0e7977#diff-2ec99f3419163ed65ed07fb36b3ac8e7a77849dec779bed388be020faba75658

This?

**Comment by frodakcin** [comment]
If you're talking about what i sent, that explanation is the live version atm

**Comment by jessechoe10** [comment]
> If you're talking about what i sent, that explanation is the live version atm

Can I still get this PR merged despite the conflict?

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 11 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:181]: ok.

**Review by frodakcin — APPROVED** [review]
Where was the merge conflict resolved?

**Comment by jessechoe10** [comment]
> Where was the merge conflict resolved?

Basically, since the original file didn't include my name at the top of the list, there was a merge conflict. I basically removed the original module authors and just added my name. :D

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 12]
```diff
@@ -1,7 +1,7 @@
 ---
 id: prefix-sums-2
 title: 'More on Prefix Sums'
-author: Darren Yao, Eric Wei, Neo Wang, Qi Wang
+author: Darren Yao, Eric Wei, Neo Wang, Jesse Choe
```
Comment (thecodingwizard) [inline content/3_Silver/Prefix_Sums_2.mdx:4]: is this (removing Qi Wang) intentional?

Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:4]: Oops no..

Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx]: I didn't even notice that :cry:

**Comment by jessechoe10** [comment]
@HarryWangATX and @maggie-j-liu can you review or approve so @frodakcin can merge?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 13]
```diff
@@ -102,11 +102,94 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array. The maximum subarray sum is the maximum possible $p[i]-p[j]$ where $j\leq i$. We can iterate over each index $i$ and keep a running minimum $p[j]$ value. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:109]: I think it would be helpful to elaborate on why this is.

**Review by maggie-j-liu — APPROVED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 13 above)
Comment (dongliu0426) [reply content/3_Silver/Prefix_Sums_2.mdx:109]: also mention that $p$ is going to be the prefix sum array

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 13 above)
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:109]:
```suggestion
Consider the prefix sum array $p$. The subarray sum over range $[i, j)$ where $j\leq i$ is $p[i]-p[j]$, so we are looking for the maximum possible value of $p[i]-p[j]$ over $0 \leq i \leq j \leq N$. 

For a fixed right bound $i$, the maximum subarray sum is

$$$
p[i]-\min_{j\leq i}{p[j]}
$$$

Thus, we can keep a running minimum to store $\min_{j\leq i}{p[j]}$ as we iterate through $i$. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
```

Is this the elaboration you were looking for?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 14]
```diff
@@ -102,11 +102,94 @@ export const problems = {
...
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:129]: lol why call it dp when its psums. ig u could but that's confusing

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 15]
```diff
@@ -102,11 +102,94 @@ export const problems = {
...
 This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+</Warning>
+
+<Spoiler title = "Alternative Solution: Kadane's Algorithm">
+
+Kadane's Algorithm finds the largest subarray sum using a greedy method. Check the video below for details:
+
+[Video explanation of the max subarray sum using Kadane's Algorithm](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s).
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:157]: nick white...

Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.mdx:157]: idk what are our opinions on putting nick white in guide LOL

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 16]
```diff
@@ -102,11 +102,94 @@ export const problems = {
...
+
+<LanguageSection>
+
+<CPPSection>
+
+C++ code:
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:163]:
```suggestion
## C++ Implementation
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 15 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:157]: I mean he explains Kadane's fairly well lol... Also, not many ppl will look at Kadane's solution anyway lol..

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 14 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:129]: Lol i'm just used to writing it as dp cuz in one editorial: http://www.usaco.org/current/data/sol_paintbarn_silver_feb19.html, they used dp to represent prefix sums so im like y not lmao

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 13 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx]: I commit @frodakcin 's change so i think i fixed...

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 15 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:165]: > nick white...

All that matters is that the approach is explained in a clear and logical way. Idt the channel really matters tbh. What matters is whether u could clearly learn, right??

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 17]
```diff
@@ -102,11 +102,102 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array $p$. The subarray sum $a_j \dots a_{i-1}$ where $j\leq i$ is $p[i]-p[j]$, so we are looking for the maximum possible value of $p[i]-p[j]$ over $0 \leq i \leq j \leq N$.
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:109]: use \texttt for variables

Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.mdx:109]: not one letter vars tho

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 16 above)
Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.mdx]: why was this resolved?

**Comment by HarryWangATX** [comment]
im guessing some latex is wrong make sure to fix that too

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 18]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
 
+### Solution
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array $p$. The subarray sum $a_j \dots a_{i-1}$ where $j\leq i$ is $p[i]-p[j]$, so we are looking for the maximum possible value of $p[i]-p[j]$ over $0 \leq i \leq j \leq N$. 
+
+For a fixed right bound $i$, the maximum subarray sum is
+
+$$$
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:113]: Why are there 3 `$`? Shouldn't 2 work?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 19]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+p[i]-\min_{j\leq i}{p[j]}
+$$$
+
+Thus, we can keep a running minimum to store $\min_{j\leq i}{p[j]}$ as we iterate through $i$. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:117]:
```suggestion
Thus, we can keep a running minimum to store $\min\limits_{j\leq i}{p[j]}$ as we iterate through $i$. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
```
move the j \leq i under \min

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 20]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+int main(){
+	int n; cin >> n;
+	vl x(n);
+	for(int i=0; i<n; i++){
+		cin >> x[i];
+	}
+	ll current_sum = x[0], max_subarray_sum = x[0];
+	for(int i=1; i<n; i++){
+		current_sum = max(current_sum + x[i], x[i]); 
+		// Continue the subarray sum or start a new subarray sum beginning at the current element
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:189]: This line is too long, maybe put a line break before "or"?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 20 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:189]: ok.

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 18 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:113]: idk i just commited what the other ppl recommended.

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 17 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:109]: Can you commit a suggestion, so that ik what you're talking ab??

Comment (maggie-j-liu) [reply content/3_Silver/Prefix_Sums_2.mdx:109]: i think the `p[i]` and `p[j]`, right?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 21]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:134]:
```suggestion
	vl pfx(n+1, 0); // Prefix sum array initially filled with 0's
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 22]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:137]:
```suggestion
		pfx[i] = pfx[i-1] + x; // Compute the prefix sum at each element
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 23]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:139]:
```suggestion
	ll max_subarray_sum = pfx[0], min_prefix_sum = pfx[0];
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 24]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+	for(int i=1; i<=n; i++){
+		// Max subarray sum is the maximum difference between two prefix sums
+		max_subarray_sum = max(max_subarray_sum, dp[i] - min_prefix_sum);
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:142]:
```suggestion
		max_subarray_sum = max(max_subarray_sum, pfx[i] - min_prefix_sum);
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 25]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+		min_prefix_sum = min(min_prefix_sum, dp[i]);
```
Comment (nwatx) [inline content/3_Silver/Prefix_Sums_2.mdx:143]:
```suggestion
		min_prefix_sum = min(min_prefix_sum, pfx[i]);
```

**Review by nwatx — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 15 above)
Comment (thecodingwizard) [reply content/3_Silver/Prefix_Sums_2.mdx:165]: I don't really like the explanation tbh, it seems geared towards an interview and not towards CP in general, so it's really long...

Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:165]: If that's the case, then should we remove the Kadane's algorithm sol?

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 20 above)
Comment (maggie-j-liu) [reply content/3_Silver/Prefix_Sums_2.mdx:189]: This hasn't been fixed yet...

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 26]
```diff
@@ -102,11 +102,102 @@ export const problems = {
...
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array $p$. The subarray sum $a_j \dots a_{i-1}$ where $j\leq i$ is $p[i]-p[j]$, so we are looking for the maximum possible value of $p[i]-p[j]$ over $0 \leq i \leq j \leq N$. 
+
+For a fixed right bound $i$, the maximum subarray sum is
+
+$$$
+p[i]-\min_{j\leq i}{p[j]}
+$$$
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:115]:
```suggestion
$$
p[i]-\min_{j\leq i}{p[j]}
$$
```

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 20 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:189]: Oops - forgot.

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 17 above)
Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.mdx:109]: yes use \texttt{p}[i] etc

**Comment by HarryWangATX** [comment]
try liveupdate

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 27]
```diff
@@ -102,11 +102,103 @@ export const problems = {
...
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:155]: why is this a warning?

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 28]
```diff
@@ -102,11 +102,103 @@ export const problems = {
...
+
 This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+</Warning>
+
+<Spoiler title = "Alternative Solution: Kadane's Algorithm">
+
+Kadane's Algorithm finds the largest subarray sum using a greedy method. Check this
+[video tutorial](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s) for more details.
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:164]: a link to wiki is alr linked so just remove the vid.

**Comment by HarryWangATX** [comment]
oops i meant request changes

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
check my comments

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 17 above)
Comment (maggie-j-liu) [reply content/3_Silver/Prefix_Sums_2.mdx]: I think `\texttt{p[i]}` looks better? idk
image

**Comment by HarryWangATX** [comment]
converted to a draft theres cus theres a lotta comments to fix for now open when ready

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 17 above)
Comment (HarryWangATX) [reply content/3_Silver/Prefix_Sums_2.mdx]: hmm i like the 2nd one but the 2nd one is standard anyways

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 27 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:155]: I thought it was better as a warning...

**Comment by jessechoe10** [comment]
> try liveupdate

I did... Somehow it looks good. Why is the PR test failing??

File: content/3_Silver/Prefix_Sums_2.mdx (hunk 28 above)
Comment (jessechoe10) [reply content/3_Silver/Prefix_Sums_2.mdx:164]: Alright... Also, the test worked earlier, but after changes, it failed :(...

**Comment by maggie-j-liu** [comment]
Something's wrong with the heading links, I think you have multiple headings with the same name (C++ Implementation)... I don't know if the website is supposed to account for this?

**Comment by thecodingwizard** [comment]
the PR is fine. The table of content generation shouldn't generate headers for stuff inside spoilers; I opened an issue #966. Multiple headings are fine :)

Feel free to merge anyway -- the bug isn't too severe

**Review by dongliu0426 — APPROVED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 29]
```diff
@@ -102,11 +102,97 @@ export const problems = {
...
+<Spoiler title="Solution">
 
-Consider the desired maximum subarray. As you go along from left to right, the prefix sum solution will mark the start of that subarray as the "current minimum prefix sum". Kadane's Algorithm, on the other hand, will set the current value to 0 at that point. As both solutions iterate through the array, they eventually find the right side of the maximum sum, and they find the answer to the problem at that location. In essence, Kadane's Algorithm stores the maximum sum of a subarray that ends at the current location (which is what the prefix sum solution calculates on each iteration), but it calculates this value greedily instead.
+Consider the prefix sum array $p$. The subarray sum $a_i \dots a_{j-1}$, where $i\leq j$ is $p[j]-p[i]$. Thus, we are looking for the maximum possible value of $p[j]-p[i]$ over $0 \leq i \leq j \leq N$. 
+
+For a fixed right bound $j$, the maximum subarray sum is
+
+$$$
+p[j]-\min_{i\leq j}{p[i]}
+$$$
```
Comment (maggie-j-liu) [inline content/3_Silver/Prefix_Sums_2.mdx:115]:
```suggestion
$$
p[j]-\min_{i\leq j}{p[i]}
$$
```

**Pull request merged**